hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -389,11 +389,14 @@ private ResponseCodeEnum zeroSumTransfers(
 					}
 				} else {
 					var adjustments = tokenTransfers.getNftTransfersList();
-					for (NftTransfer adjustment : adjustments) {
-						validity = adjustTokenBalance(adjustment.getSenderAccountID(), adjustment.getReceiverAccountID(),
-								id, adjustment.getSerialNumber());
-						if (validity != OK) {
-							break;
+					validity = checkMaxNftTransfersLen(adjustments.size());
```

Comment:
This should be a precheck in `CryptoTransfersTransitionLogic` -> `validate(TransactionBody txn)` method

Reply:
It has been moved